src/app.ts

```diff
@@ -1,26 +1,120 @@
+import { Command } from "commander";
 import { CodeTokenizer } from "./lib/codeTokenizer";
-import { Comparison } from "./lib/comparison";
-import { Summary } from "./lib/summary";
+import { Comparison, ComparisonFilterOptions } from "./lib/comparison";
+import { Matches } from "./lib/comparison.js";
+import { FilterOptions, Summary } from "./lib/summary.js";
+
+let locations: string[] = [];
+const program = new Command();
+
+// Initial program description and version
+program.version("0.0.1").description("Plagiarism detection for programming exercises");
+
+program
+  .option(
+    "-l, --language <language>",
+    "Programming language used in the submitted files.",
+    "javascript",
+  )
+  .option(
+    "-b, --base <base>",
+    "Specifies a base file. Matches with code from this file will never be reported in the output. A typical base " +
+      "file is the supplied code for an exercise.",
+  )
+  .option(
+    "-m, --maximum-fragment-count <number>",
+    "The -m option sets the maximum number of times a given fragment may appear before it is ignored. A code fragment" +
+      "that appears in many programs is probably legitimate sharing and not the result of plagiarism. With -m N " +
+      "any fragment appearing in more than N program is filtered out. This option has precedence over the -M option, " +
+      "which is set to 0.9 by default.",
+  )
+  .option(
+    "-M --maximum-fragment-percentage <float>",
+    "The -M option sets how many percent of the files the code fragment may appear before it is ignored. A fragment " +
+      "of code that appears in many programs is probably legitimate sharing and not the result of plagiarism. With " +
+      "-M N any fragment appearing in more than N percent of the files is filtered out. " +
+      "Must be a value between 0 and 1.",
+    0.9,
+  )
+  .option("-c, --comment <string>", "Comment string that is attached to the generated report")
+  .option(
+    "-n, --fragment-output-limit",
+    "Specifies how many matching fragment are shown in the result. All fragment are " +
+      "shown then this option isn't used.",
+  )
+  .option(
+    "-s, --minimum-lines-shortest <integer>",
+    "The minimum amount of lines in the shortest code fragment in a comparison before it is shown",
+    1,
+  )
+  .option(
+    "-S, --minimum-lines-longest <integer>",
+    "The minimum amount of lines in the longest code fragment in a comparison before it is shown",
+    1,
+  )
+  .option(
+    "-g, --maximum-gap-size <integer>",
+    "If two fragment need to be joined, then this parameter specifies how large the gap between the two fragment may" +
```

Comment:
"If two fragments are close to each other, they will be merged into a single fragment if the gap between them is smaller than the given number of lines."